### PR TITLE
Added a method to remove multiple empty lines.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
## Summary by CodeRabbit

- **New Features**
  - Introduced a utility to collapse multiple consecutive empty lines in text, preserving the original line ending style.
- **Documentation**
  - Updated documentation to describe the new empty line collapsing feature and its usage.
  - Added guidelines on code linting and standards, including best practices for exception handling and variable naming.
- **Bug Fixes**
  - Improved internal handling of regular expressions and null checks to prevent potential errors.
- **Tests**
  - Added comprehensive tests for the new empty line collapsing feature, covering various edge cases and line endings.
  - Renamed and clarified test methods for better readability.
- **Chores**
  - Expanded static analysis and code linting coverage to include additional files.

### CODE REVIEW

File: File.php [hunk 1]
```diff
@@ -737,13 +753,109 @@
...
+            return $content;
+          }
+        }
+        else {
+          // For pure empty lines: collapse all consecutive newlines to single.
+          $normalized = preg_replace("/\n{2,}/", "\n", $normalized);
+          if ($normalized === NULL) {
+            return $content;
+          }
+        }
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply File.php:834]: the rules are
1. if there is 1+ new CRLF (or any variant of it) at the end - it must be only one source variant left.
2. 2+ CRLF (or any variant of it) within the content are collapsed into exactly 2.
3. any CRLF should be stripped from the beginning, but only if there is content.
4. string not ending with CRLF (or any variant of it) should not have the CRLF added.

[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply File.php:834]: @coderabbitai
do you have a suggestion for improving the method code to simplify it?
